packages/flutter_tools/lib/src/observatory.dart

```diff
@@ -113,10 +115,17 @@ class Observatory {
     });
   }
 
-  Future<Response> isolateReload(String isolateId) {
-    return sendRequest('isolateReload', <String, dynamic>{
-      'isolateId': isolateId
-    });
+  Future<bool> reloadSources(String isolateId) async {
+    Future<Event> future = onIsolateEvent
+      .where((Event event) => event.kind == 'IsolateReload')
+      .single;
+
+    await sendRequest('_reloadSources', <String, dynamic>{ 'isolateId': isolateId });
+    Event event = await future.timeout(new Duration(seconds: 20));
```

Comment:
It's not too bad with a `Completer`:

```dart
  Completer<Event> whenIsolateReloads = new Completer<Event>();
  StreamSubscription sub = onIsolateEvent
    .where((Event event) => event.kind == 'IsolateReload')
    .listen(whenIsolateReloads.complete);

  try {
    Event event = await whenIsolateReloads.future.timeout(new Duration(seconds: 20));
  } finally {
    await sub.cancel();
  }
```

Up to you. Doesn't matter if this is rare.